### PR TITLE
DSND-548: Add non 2xx response codes

### PR DESCRIPTION
**Pull request by iichr**
- Return non-2XX response codes when issues/errors occur when calling the GET and PATCH endpoints
- Improved logging

Resolves:
https://companieshouse.atlassian.net/browse/DSND-548

### CODE REVIEW

**Review by kgeddis-ch — DISMISSED** [review]
Looks good to me :)

File: src/main/java/uk/gov/companieshouse/company/profile/converter/JsonDateTimeConverter.java [hunk 1]
```diff
@@ -25,6 +30,7 @@ public void convert(Long value, StrictJsonWriter writer) {
             String formattedDate = DATE_TIME_FORMATTER.format(instant);
             writer.writeString(formattedDate);
         } catch (Exception ex) {
+            LOGGER.error(ex.getMessage());
```
Comment (gseregni-ch) [inline src/main/java/uk/gov/companieshouse/company/profile/converter/JsonDateTimeConverter.java]: Can you please add a meaningful message in front of the exception?

File: src/main/java/uk/gov/companieshouse/company/profile/controller/CompanyProfileController.java [hunk 2]
```diff
@@ -35,7 +35,6 @@ public CompanyProfileController(CompanyProfileService companyProfileService) {
         return companyProfileService.get(companyNumber)
                 .map(document ->
```
Comment (zahidm3) [inline src/main/java/uk/gov/companieshouse/company/profile/controller/CompanyProfileController.java:36]: Hi, I dont see where we handling default spring de-serialisation exceptions if we send invalid json in payload from client?

**Review by gseregni-ch — APPROVED** [review]

File: src/main/java/uk/gov/companieshouse/company/profile/controller/CompanyProfileController.java (hunk 2 above)
Comment (iichr) [reply src/main/java/uk/gov/companieshouse/company/profile/controller/CompanyProfileController.java:36]: Should occur during `save` in `updateInsolvencyLink`, I've also just added a small update to throw bad request when generating the update query - there's currently multiple points of failure. This whole method will be reworked as part of a ticket you've raised as well so the error handling will also evolve :)

File: src/main/java/uk/gov/companieshouse/company/profile/converter/JsonDateTimeConverter.java (hunk 1 above)
Comment (iichr) [reply src/main/java/uk/gov/companieshouse/company/profile/converter/JsonDateTimeConverter.java]: Yes, sure

**Review by gseregni-ch — APPROVED** [review]

**Review by kgeddis-ch — APPROVED** [review]

File: src/main/java/uk/gov/companieshouse/company/profile/controller/CompanyProfileController.java (hunk 2 above)
Comment (iichr) [reply src/main/java/uk/gov/companieshouse/company/profile/controller/CompanyProfileController.java:36]: Clarified further - we do throw a 400 - intercepting HttpMessageNotReadableException in Controller advice!

**Pull request merged**